covid_moonshot_ml/nn/loss/MSELoss.py

```diff
@@ -0,0 +1,113 @@
+import torch
+from torch.nn.modules.loss import MSELoss as TorchMSELoss
+
+
+class MSELoss(TorchMSELoss):
+    def __init__(self, loss_type=None):
+        """
+        Class for calculating MSE loss, with various options
+
+        Parameters
+        ----------
+        loss_type : str, optional
+            Which type of loss to use:
+             * None: vanilla MSE
+             * "step": step MSE loss
+             * "uncertainty": MSE loss with added uncertainty
+        """
+        ## No reduction so we can apply whatever adjustment to each sample
+        super(MSELoss, self).__init__(reduction="none")
+
+        if loss_type is not None:
+            loss_type = loss_type.lower()
+            if loss_type == "step":
+                self.loss_function = self.step_loss
+            elif loss_type == "uncertainty":
+                self.loss_function = self.uncertainty_loss
+            else:
+                raise ValueError(f'Unknown loss_type "{loss_type}"')
+        else:
+            self.loss_function = super(MSELoss, self).forward
+
+        self.loss_type = loss_type
+
+    def forward(self, input, target, in_range, uncertainty):
+        """
+        Dispatch method for calculating loss. All arguments should be passed
+        regardless of actual loss function to keep an identical signature for
+        this class. Data is passed to `self.loss_function`.
+        """
+
+        if self.loss_type is None:
+            return self.loss_function(input, target).mean()
+        elif self.loss_type == "step":
+            return self.loss_function(input, target, in_range)
```

Comment:
Can you document that this calls `self.loss_function = self.step_loss`?

Reply:
done in [4fbf3c0](https://github.com/choderalab/covid-moonshot-ml/pull/33/commits/4fbf3c0e4f00b7bc3f65413484f7cbdb9987003c)